src/OrchardCore.Modules/OrchardCore.Https/Startup.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Rewrite;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.Environment.Navigation;
+using OrchardCore.Https.Drivers;
+using OrchardCore.Https.Services;
+using OrchardCore.Modules;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Https
+{
+    public class Startup : StartupBase
+    {
+        public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var rewriteOptions = new RewriteOptions();
```

Comment:
That looks complicated, maybe you are right but I would have expected it to be simpler. Like just resolving the `HttpsSettings` here and registering the rewrite rule right there.

Reply:
Yeah I thought it would be just like MvcOptions and all the OpenId Options, where any registered IConfigureOptions would be invoked. Maybe a bug on ASP.NET Core Rewrite?